Tests/Utilities/ResultTests.swift

```diff
@@ -0,0 +1,87 @@
+import Nimble
+import Quick
+
+@testable import Zomavan
+
+final class ResultTests: QuickSpec {
+
+    override func spec() {
+
+        describe("a Result") {
+
+            var subject: Result<Int, URLError>!
+
+            afterEach {
+                subject = nil
+            }
+
+            context("that represents a success") {
+
+                beforeEach {
+                    subject = .success(10)
+                }
+
+                it("captures a value") {
+
+                    var value: Int?
+
+                    if case let .success(capturedValue)? = subject {
```

Comment:
I notice that it's the same as below with the "failure" case. What do you think about implementing accessors for these so that we don't need this complex statement here?

```swift
var value: Value? {
    if case let .success(value) = self {
        return value
    }
}
```

Reply:
I do think that it would be worthwhile. Until we see uses for this in the codebase, what do you think about leaving it undone and then creating them as we need?

For the scope of this work, I think we may not use the described accessors that much.

Reply:
Sounds fair to me. I'll keep an eye out to review tests for those when/if implemented!